src/sundials/sundials_stepper.c

```diff
@@ -0,0 +1,171 @@
+
+
+#include <stdlib.h>
+#include <string.h>
+#include <sundials/priv/sundials_errors_impl.h>
+#include <sundials/sundials_core.h>
+
+#include "sundials/sundials_errors.h"
+#include "sundials/sundials_nvector.h"
+#include "sundials/sundials_types.h"
+#include "sundials_stepper_impl.h"
+
+SUNErrCode SUNStepper_Create(SUNContext sunctx, SUNStepper* stepper_ptr)
+{
+  SUNFunctionBegin(sunctx);
+  SUNCheck(stepper_ptr, SUN_ERR_ARG_CORRUPT);
+
+  SUNStepper stepper = malloc(sizeof(*stepper));
+  SUNAssert(stepper, SUN_ERR_MALLOC_FAIL);
+
+  stepper->content   = NULL;
+  stepper->sunctx    = sunctx;
+  stepper->last_flag = SUN_SUCCESS;
+
+  stepper->ops = malloc(sizeof(*(stepper->ops)));
+  SUNAssert(stepper->ops, SUN_ERR_MALLOC_FAIL);
+
+  stepper->ops->evolve      = NULL;
+  stepper->ops->onestep     = NULL;
+  stepper->ops->fullrhs     = NULL;
+  stepper->ops->reset       = NULL;
+  stepper->ops->setstoptime = NULL;
+  stepper->ops->setforcing  = NULL;
+
+  *stepper_ptr = stepper;
+
+  return SUN_SUCCESS;
+}
+
+SUNErrCode SUNStepper_Destroy(SUNStepper* stepper_ptr)
+{
+  if (stepper_ptr != NULL)
+  {
+    free((*stepper_ptr)->ops);
+    free(*stepper_ptr);
+    *stepper_ptr = NULL;
+  }
```

Comment:
We should probably allow the user to attach their own `Destroy` implementation to free any stepper content